app/services/revocation_registry.py

```diff
@@ -309,61 +311,143 @@ async def revoke_credential(
+            f"Failed to publish pending revocations: {e.reason}.", e.status
         ) from e
-    async with get_governance_controller() as endorser_controller:
-        logger.info("Endorsing what is presumed to be a revocation transaction")
-        await endorser_controller.endorse_transaction.endorse_transaction(
-            tran_id=txn_record["transaction_id"]
+
+    bound_logger.info("Successfully published pending revocations.")
+
+
+async def clear_pending_revocations(
+    controller: AcaPyClient, purge: ClearPendingRevocationsRequest
+) -> PublishRevocations:
+    """
+        Clear pending revocations
+
+    Args:
+        controller (AcaPyClient): aca-py client
+        purge (ClearPendingRevocationsRequest): The revocation registry ID.
+
+    Raises:
+        Exception: When the pending revocations could not be cleared
+
+    """
+    bound_logger = logger.bind(body=purge)
+
+    bound_logger.info("Validating revocation registry ids")
+    await validate_rev_reg_ids(controller=controller, request=purge)
+
+    try:
+        result = await controller.revocation.clear_pending_revocations(
+            body=ClearPendingRevocationsRequest(purge=purge)
         )
-    logger.info("Successfully endorsed transaction of revocation.")
+    except ApiException as e:
+        bound_logger.exception(
+            "An ApiException was caught while clearing pending revocations. The error message is: '{}'.",
+            e.reason,
+        )
+        raise CloudApiException(
+            f"Failed to clear pending revocations: {e.reason}.", e.status
+        ) from e
+
+    bound_logger.info("Successfully cleared pending revocations.")
+    return result
+
+
+async def get_credential_revocation_record(
+    controller: AcaPyClient,
+    credential_exchange_id: str,
+    credential_revocation_id: str,
+    revocation_registry_id: str,
+) -> IssuerCredRevRecord:
+    """
+        Get the revocation status for a credential
+
+    Args:
+        controller (AcaPyClient): aca-py client
+        credential_exchange_id (str): The credential exchange ID.
+        credential_revocation_id (str): The credential revocation ID.
+        revocation_registry_id (str): The revocation registry ID.
+
+    Raises:
+        Exception: When failed to get revocation status.
+
+    """
+    bound_logger = logger.bind(
+        body={
+            "credential_exchange_id": credential_exchange_id,
+            "credential_revocation_id": credential_revocation_id,
+            "revocation_registry_id": revocation_registry_id,
+        }
+    )
+    bound_logger.info("Fetching the revocation status for a credential exchange")
+
+    try:
+        result = await controller.revocation.get_revocation_status(
+            cred_ex_id=credential_exchange_id if credential_exchange_id else None,
+            cred_rev_id=credential_revocation_id if credential_revocation_id else None,
+            rev_reg_id=revocation_registry_id if revocation_registry_id else None,
```

Comment:
I think the if else None is not necessary, because the routes methods have default Nones. Just update the method inputs to:
```py
async def get_credential_revocation_record(
    controller: AcaPyClient,
    credential_exchange_id: Optional[str] = None,
    credential_revocation_id: Optional[str] = None,
    revocation_registry_id: Optional[str] = None,
) -> IssuerCredRevRecord:
```